rudder-lang/src/ir/value.rs

```diff
@@ -161,24 +171,44 @@ impl<'src> Value<'src> {
     }
 
     pub fn context_check(&self, context: &VarContext<'src>) -> Result<()> {
-        match self {
-            Value::String(s) => {
-                map_results(s.data.iter(), |e| match e {
-                    PInterpolatedElement::Static(_) => Ok(()),
-                    PInterpolatedElement::Variable(_v) => Ok(()),
-                    // TODO
-                    //                            match getter(Token::new("", v)) {
-                    //                                None => fail!(s.pos, "Variable {} does not exist at {}", v, s.pos),
-                    //                                _ => Ok(()),
-                    //                            },
-                })
+        let exist = |tk| -> Result<()> {
```

Comment:
leaving this unimplemented! disallowed usage of different types of variables in rudderlang although they are properly translated into strings. Floats/Int can't hld interpolated values so I dont think any context check is important here